tests/test_testing.py

```diff
@@ -27,7 +74,49 @@ class SimpleHttpApp(AsyncConsumer):
         })
 
 
-@pytest.mark.asyncio
+async def test_auth_communicator(user, scope):
+    """
+    Tests that the authentication communicator methods work.
+    """
+    communicator = AuthCommunicator(SimpleHttpApp, scope, user=user)
+    assert communicator.scope["user"] is user
+    assert "session" in communicator.scope
+    assert await get_user(communicator.scope) == AnonymousUser()
+    assert communicator.scope is communicator.instance.scope
```

Comment:
Do you think it was better to copy the scope before passing it to the application in order to be more consistent? (It's another scope when you use middleware as you can see below.)

I think I'll bring these two tests together into one when this is decided.

Reply:
I don't think you should be using an `is` comparison here, anyway, copying or not.